src/Components/CreateImageWizardV2/ImageWizardContext.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from 'react';
+
+import {
+  EnvironmentStateType,
+  filterEnvironment,
+  useGetAllowedTargets,
+} from './steps/ImageOutput/Environment';
+import { GCPAccountTypes } from './steps/TargetEnvironment/GCP/GCPTarget';
+import { useGetAccountData } from './steps/TargetEnvironment/SourcesSelect';
+
+import { RHEL_9, X86_64 } from '../../constants';
+import { ArchitectureItem, Distributions } from '../../store/imageBuilderApi';
+
+type ImageWizardContextPropType = {
+  releaseState: [
+    Distributions,
+    React.Dispatch<React.SetStateAction<Distributions>>
+  ];
+  architectureState: [
+    ArchitectureItem['arch'],
+    React.Dispatch<React.SetStateAction<ArchitectureItem['arch']>>
+  ];
+  environmentState: [
+    EnvironmentStateType,
+    React.Dispatch<React.SetStateAction<EnvironmentStateType>>
+  ];
+  isAwsManualState: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
+  awsSourceState: [
+    [number, string],
+    React.Dispatch<React.SetStateAction<[number, string]>>
+  ];
+  associatedAwsAccountIdState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+  isShareGoogleAccountState: [
+    boolean,
+    React.Dispatch<React.SetStateAction<boolean>>
+  ];
+  gcpAccountTypeState: [
+    GCPAccountTypes,
+    React.Dispatch<React.SetStateAction<GCPAccountTypes>>
+  ];
+  gcpAccountEmailState: [string, React.Dispatch<React.SetStateAction<string>>];
+  gcpDomainState: [string, React.Dispatch<React.SetStateAction<string>>];
+  isAzureManualState: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
+  azureSourceState: [
+    [number, string],
+    React.Dispatch<React.SetStateAction<[number, string]>>
+  ];
+  azureTenantIdState: [string, React.Dispatch<React.SetStateAction<string>>];
+  azureSubscriptionIdState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+  azureResourceGroupState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+};
+
+export const ImageWizardContext =
+  React.createContext<ImageWizardContextPropType>({
+    releaseState: [RHEL_9, () => {}],
```

Comment:
The `-State` suffix on these variables seems redundant to me... we know it is the state, don't we? We could cut down on verbosity a bit by removing it.

Reply:
The reason why there's `State` everywhere is because the state and its setter are packed together. And then you unpack them the same way you would with `useState`. But I see that you already don't like this packing, so why not have just the state and its setter together in the context.